lib/src/core/IntersectionAndProjection/geodesic_lines.dart

```diff
@@ -0,0 +1,125 @@
+import '../core.dart';
+
+///  Geodesic lines
+class GeodesicLines {
+  /// CalculateGeodesicLineIntersection function, which takes the starting
+  /// and ending points of two geodesic lines as input and calculates their
+  ///  intersection point using various geodesic formulas.
+  /// The function returns a LatLng representing the intersection point or
+  ///  null if no intersection is found.
+  static LatLng? calculateGeodesicLineIntersection(
+      LatLng start1, LatLng end1, LatLng start2, LatLng end2) {
+    const double earthRadius = 6371000; // Earth's radius in meters
```

Comment:
radius seems to be used in different files. can we put it to a constant file?

Reply:
I wanna avoid global variables... I don't like the code that is mutable...